### DISCUSSION TITLE
Do "picked localized routes" result in prerendered routes?

### DISCUSSION THREAD

**Discussion by nachoadjust** [nuxt-modules/i18n#2564] [Q&A]
Hello! 👋

I was wondering if setting the localized routes in pages as [shown in the docs](https://i18n.nuxtjs.org/guide/ignoring-localized-routes#pick-localized-routes) result in these routes being added into `routeRules` as "prerender"?

---
**Objective:**
I am attempting to statically generate (now called [prerender](https://nuxt.com/docs/guide/concepts/rendering#route-rules?) I believe?) the routes defined in each page definition of `defineI18nRoute`.

My assumption was that if I declare `defineI18nRoute` in a page:

```js
// ./pages/blog/index.vue
defineI18nRoute({ locales: ['en', 'ja', 'zh'] })
...
```

Later those routes would get "auto-injected" into `routeRules` as prerendered routes. 

Something like:
```js
// nuxt.config.ts
export default defineNuxtConfig({
  ...,
  routeRules: {
    '/**': { isr: 60 },
    '/': { prerender: true },  
    ...[LOCALIZED ROUTES ARE AUTOMAGICALLY INJECTED HERE]
  }
})
```
---

**What I got so far:** 

- I see the localized routes generate as expected if I run `nuxt generate` (they show in the build logs) 
   but they don't when done via `nuxt build` (they don't show in the build logs). 
   In my case my project is built using `nuxt build`.

- After running `nuxt build` > in runtime: tho the localized routes are not prerendered I see things behave as expected.
    A page 404s if the locale in the url is not matching any allowed value in that page's `defineI18nRoute`.
    ```sh
    mysite.com/ja/blog/ # works!
    mysite.com/es/blog/ # 404 (which is expected, cos we only allowed 'en', 'ja', 'zh')
    ```

Hope all of this makes sense!
If anyone could help me understand the behaviour I should expect from localized routes when doing `nuxt build` and if it makes sense to prerender those routes (and how 😅)  I'd be very grateful!

Thanks in advance!!

**Comment by FranckFreiburger** [comment]
Hello, I also wondering how to prerender localized routes.
